yt_dlp/extractor/brightcove.py

```diff
@@ -577,6 +577,22 @@ def build_format_id(kind):
         if duration is not None and duration <= 0:
             is_live = True
 
+        common_res = ["160x90", "320x180", "480x270", "640x360", "768x432", "1024x576", "1280x720", "1366x768", "1920x1080"]
+        thumbnails = []
+        if '640x360' in json_data.get('poster'):
+            base_url = json_data.get('poster')
+            replace = '640x360'
+        else:
+            base_url = json_data.get('thumbnail')
+            replace = '160x90'
+        for res in common_res:
+            thumbnails.append({
+                'url': base_url.replace(replace, res),
```

Comment:
much easier to do `re.sub(r'\d+x\d+', ...)` on `dict_get(json_data, ('poster', 'thumbnail')`